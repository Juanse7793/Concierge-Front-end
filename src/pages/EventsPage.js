import { useDispatch, useSelector } from 'react-redux';
import { deleteEvent } from '../redux/reducers/events';

const EventsPage = () => {
  const events = useSelector((state) => state.events);
  const dispatch = useDispatch();

  const deleteEventHandler = (e) => {
    const id = e.target.getAttribute('id');
    dispatch(deleteEvent(id));
  };

  return (
    <div>
      {events.map((event) => (
        <ul key={event.id}>
          <li id="event-name">
            <p>{event.name}</p>
            <button type="button" className="delete-btn" id={event.id} onClick={deleteEventHandler}>Delete</button>
          </li>
        </ul>
      ))}
    </div>
  );
};

export default EventsPage;
