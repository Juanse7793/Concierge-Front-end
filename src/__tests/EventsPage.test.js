import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import EventsPage from '../pages/EventsPage';

const mockStore = configureMockStore();
const store = mockStore({
  user: { user: { id: 1, name: 'John' } },
  events: {
    events: [
      {
        id: 1,
        name: 'foo',
        start_date: '2022-01-01',
        end_date: '2023-01-01',
      },
    ],
    adding: false,
  },
});

it('matches snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <EventsPage />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
