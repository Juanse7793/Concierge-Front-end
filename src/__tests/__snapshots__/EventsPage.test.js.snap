// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section>
  <ul>
    <li
      className="row pill list"
    >
      <p>
        foo
      </p>
      <div
        className="row"
      >
        <small>
          2022-01-01 ~ 2023-01-01
        </small>
        <button
          className="red pill link"
          id={1}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </li>
  </ul>
</section>
`;
